filesystem/iso9660.ksy

```diff
@@ -7,269 +7,962 @@ meta:
+          - id: loc_opt_m_path_table
+            doc-ref: ecma-119 8.5.11
+            type: u4be
+          - id: directory_record_for_root_directory
+            doc-ref: ecma-119 8.5.12
+            doc: |
+              The root directory_record with body is 34 bytes.
+              In other cases the size is variable.
+            type: directory_record
+            size: 0x22
+          - id: volume_set_identifier
+            doc-ref: ecma-119 8.5.13
+            type: text128
+          - id: publisher_identifier
+            doc-ref: ecma-119 8.5.14
+            type: text128
+          - id: data_preparer_identifier
+            doc-ref: ecma-119 8.5.15
+            type: text128
+          - id: application_identifier
+            doc-ref: ecma-119 8.5.16
+            type: text128
+          - id: copyright_file_identifier
+            doc-ref: ecma-119 8.5.17
+            type: text37
+          - id: abstract_file_identifier
+            doc-ref: ecma-119 8.5.18
+            type: text37
+          - id: bibliographic_file_identifier
+            doc-ref: ecma-119 8.5.19
+            type: text37
+          - id: volume_creation_date_and_time
+            doc-ref: ecma-119 8.5
+            type: datetime_long
+          - id: volume_modification_date_and_time
+            doc-ref: ecma-119 8.5
+            type: datetime_long
+          - id: volume_expiration_date_and_time
+            doc-ref: ecma-119 8.5
+            type: datetime_long
+          - id: volume_effective_date_and_time
+            doc-ref: ecma-119 8.5
+            type: datetime_long
+          - id: file_structure_version
+            doc-ref: ecma-119 8.5
+            type: s1
+        instances:
+          path_table:
+            io: _root._io
+            pos: _root.sector_size * loc_l_path_table
+            size: path_table_size.le
+            type: path_table_records
+      path_table_records:
+        doc-ref: ecma-119 9.4
+        seq:
+          - id: path_table_record
+            type: path_table_record
+            repeat: eos
+        types:
+          path_table_record:
+            doc-ref: ecma-119 9.4
+            seq:
+              - id: len_dir_name
+                doc-ref: ecma-119 9.4.1
+                type: u1
+              - id: len_ext_rec
+                doc-ref: ecma-119 9.4.2
+                type: u1
+              - id: loc_ext
+                doc-ref: ecma-119 9.4.3
+                type: u4le
+              - id: parent_dir_num
+                doc-ref: ecma-119 9.4.4
+                type: u2le
+              - id: dir_name
+                doc-ref: ecma-119 9.4.5
+                type: str
+                size: len_dir_name
+              - id: padding_field
+                doc-ref: ecma-119 9.4.6
+                doc: |
+                  Padding field is added when len_dir_name contains an odd number
+                type: u1
+                valid: 0
+                if: len_dir_name % 2 == 1
```

Comment:
I don't consider this a good idea - if there is a padding field and all you can say about it is that it's filled with 0's, it should use a byte array (and therefore go with `contents`):

```suggestion
              - id: padding_field
                doc-ref: ecma-119 9.4.6
                doc: |
                  Padding field is added when len_dir_name contains an odd number
                contents: [ 0x0 ]
                if: len_dir_name % 2 == 1
```

The difference from the `version` field (https://github.com/kaitai-io/kaitai_struct_formats/pull/82#discussion_r606848014) is in the intent. You want `version` to be an integer, because it really is a meaningful number. On the other hand, magic signatures and paddings should be just byte arrays.

Reply:
fixed

Reply:
> fixed

This also needs to be done on other padding fields:

https://github.com/kaitai-io/kaitai_struct_formats/blob/3188df4e7a0fe8f834175d7415591f72a455c62c/filesystem/iso9660.ksy#L467-L473

https://github.com/kaitai-io/kaitai_struct_formats/blob/3188df4e7a0fe8f834175d7415591f72a455c62c/filesystem/iso9660.ksy#L158-L161

Reply:
fixed